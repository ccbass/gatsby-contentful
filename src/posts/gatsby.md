---
title: "Basic Gatsby Tutorial"
date: "2021-08-28"
---

Here's the tutorial markdown post!

## Topics

1. Gatsby
2. GraphQL
3. React
