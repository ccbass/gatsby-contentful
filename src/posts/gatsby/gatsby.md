---
title: "Basic Gatsby Tutorial"
date: "2021-08-28"
---

Here's the tutorial markdown post!

![grass image](./grass.jpg)
## Topics

1. Gatsby
2. GraphQL
3. React
