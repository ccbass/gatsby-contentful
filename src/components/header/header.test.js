import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import Header from "./header"


beforeEach(() => {
  jest.clearAllMocks();

  useStaticQuery.mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: `Full-Stack Engineering Blog`,
        },
      },
  }))
})

describe("Header Tests", () =>
  it("Renders as expected vs snapshot.", () => {
    const header = renderer.create(<Header />).toJSON()
    expect(header).toMatchSnapshot()

  }))
