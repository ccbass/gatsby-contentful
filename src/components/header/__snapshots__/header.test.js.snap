// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Renders as expected vs snapshot. 1`] = `
<header>
  <h1>
    <a
      href="/"
    >
      Full-Stack Engineering Blog
    </a>
  </h1>
  <nav>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
